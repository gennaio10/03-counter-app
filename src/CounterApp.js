
import React, { useState } from 'react';
import PropTypes from 'prop-types';

const CounterApp = ( { value = 0 } ) => {

    const [ counter, setCounter ] = useState(value);

    const handleAdd = () => setCounter(counter + 1);

    const handleSubtract = () => setCounter(counter - 1);

    const handleReset = () => {
        setCounter(value);
    };

    const handleTestClick = (e) => {
        console.log(e);
    };

    return (
        <>
            <h1>CounterApp</h1>
            <h2 id="valorcontador">{ counter }</h2>
            <button id="btnadd" onClick={ handleAdd }>+1</button>
            <button id="btnreset" onClick={ handleReset }>Reset</button>
            <button id="btnsub" onClick={ handleSubtract }>-1</button>
            <br></br>
            <button onClick={ handleTestClick }>Click Event, paso de e por default</button>
        </>
    );
};

CounterApp.propTypes = {
    value: PropTypes.number,
};

export default CounterApp;
