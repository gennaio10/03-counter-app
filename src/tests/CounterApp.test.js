import React from "react";
import "@testing-library/jest-dom";
import CounterApp from "../CounterApp";
import { shallow } from "enzyme";

describe("Pruebas de CounterApp.js", () => {
  test("debe mostrar el valor deseado", () => {
    const valorDeseado = 10;
    const wrapperLocal = shallow(<CounterApp value={valorDeseado} />);
    expect(wrapperLocal).toMatchSnapshot();
  });

  let wrapper = shallow(<CounterApp />);

  beforeEach(() => {
    wrapper = shallow(<CounterApp />);
  });

  test("debe mostrar el valor por defecto", () => {
    const valorDefecto = "0";
    const valorEncontrado = wrapper.find("#valorcontador").text().trim();
    expect(valorEncontrado).toBe(valorDefecto);
  });

  test("debe decrementar el contador en +1 con el boton +1", () => {
    const btnaddEncontrado = wrapper.find("#btnadd");
    btnaddEncontrado.simulate("click");
    const valorDeseado = "1";
    const valorEncontrado = wrapper.find("#valorcontador").text().trim();
    expect(valorEncontrado).toBe(valorDeseado);
  });

  test("debe incrementar el contador en -1 con el boton -1", () => {
    const btnaddEncontrado = wrapper.find("#btnsub");
    btnaddEncontrado.simulate("click");
    const valorDeseado = "-1";
    const valorEncontrado = wrapper.find("#valorcontador").text().trim();
    expect(valorEncontrado).toBe(valorDeseado);
  });

  test("debe dejar al valor por defecto con el boton reset", () => {
    wrapper.find("#btnadd").simulate("click");
    wrapper.find("#btnreset").simulate("click");
    const valorDeseado = "0";
    const valorEncontrado = wrapper.find("#valorcontador").text().trim();
    expect(valorEncontrado).toBe(valorDeseado);
  });
});
