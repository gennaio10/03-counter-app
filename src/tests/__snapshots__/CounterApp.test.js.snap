// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de CounterApp.js debe mostrar el valor deseado 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2
    id="valorcontador"
  >
    10
  </h2>
  <button
    id="btnadd"
    onClick={[Function]}
  >
    +1
  </button>
  <button
    id="btnreset"
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    id="btnsub"
    onClick={[Function]}
  >
    -1
  </button>
  <br />
  <button
    onClick={[Function]}
  >
    Click Event, paso de e por default
  </button>
</Fragment>
`;
